### ISSUE TITLE
PyTorch 1.13.0 conda install failed during import due to undefined symbol

### ISSUE THREAD

**Issue by roywei** [pytorch/pytorch#90673]
### 🐛 Describe the bug

conda installed pytorch 1.13.0 will fail during import

```
conda install pytorch torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia
```
```
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/fsx/conda/envs/test_libcusparse/lib/python3.10/site-packages/torch/__init__.py", line 192, in <module>
    from torch._C import *  # noqa: F403
ImportError: /fsx/conda/envs/test_libcusparse/lib/python3.10/site-packages/torch/lib/libtorch_cuda_cu.so: undefined symbol: cusparseSpSM_analysis, version libcusparse.so.11
>>>
```

### Root cause
libcusparse is probably loosely pined in conda builder
https://github.com/pytorch/builder/tree/main/conda
https://anaconda.org/nvidia/libcusparse/labels

since pytorch conda channel made nvidia cuda as a dependency, there is an update of libcusparse 2 days ago that will be downloaded during pytorch install and will result in an undefined symbol. 
libcusparse should be pined at version 11.7.3.50 instead of latest 12.0.0.76

what's been downloaded from conda install command.
+ libcusparse                12.0.0.76  0                             nvidia/linux-64         Cached
+ libcusparse-dev            12.0.0.76  0                             nvidia/linux-64         Cached

what should be used for cuda 11.7 and 11.7.1

cuda-11.7.0 | 11.7.3.50
-- | --
cuda-11.7.1 | 11.7.4.91

### temp workaround
manually install old version works `conda install libcusparse=11.7.3.50 -c nvidia`

### Versions

1.13.0

cc @ezyang @seemethere @malfet

**Comment by roywei** [comment]
tried another clean install, this time it shows libcudart undefined symbo. Not really sure about the root cause now.

```
conda clean --all -f
conda install pytorch torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia
```

```
ype "help", "copyright", "credits" or "license" for more information.
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/fsx/conda/lib/python3.9/site-packages/torch/__init__.py", line 192, in <module>
    from torch._C import *  # noqa: F403
ImportError: /fsx/conda/lib/python3.9/site-packages/torch/lib/libtorch_cuda_cpp.so: undefined symbol: cudaGraphRetainUserObject, version libcudart.so.11.0
```

**Comment by jphdotam** [comment]
I'm also getting this error doing a clean conda install of the nightly build (i.e. Pytorch 2.0) using the command recommended on the Pytorch website (ie `conda install pytorch torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia`) on Linux.

```
Python 3.10.8 (main, Nov 24 2022, 14:13:03) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/james/miniconda3/envs/mammo2/lib/python3.10/site-packages/torch/__init__.py", line 219, in <module>
    from torch._C import *  # noqa: F403
ImportError: /home/james/miniconda3/envs/mammo2/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so: undefined symbol: cudaGraphInstantiateWithFlags, version libcudart.so.11.0
```

**Comment by atalman** [comment]
This should be resolved now by the PR: https://github.com/pytorch/builder/pull/1227
Deployed to nightly/test and prod

**Comment by malfet** [comment]
Thank you for reporting. This regression happened because CUDA-12.0 got published to nvidia anaconda channel, and `pytorch-cuda` package did not have sufficient guards for it. Fixed right now, please do not hesitate to reopen/file a new issue if this still persists.

**Comment by atalman** [comment]
There is still an issue in production, repodata refresh.: https://github.com/ContinuumIO/anaconda-issues/issues/13108
nightlies and test are fixed

**Comment by roywei** [comment]
I tested a few hours ago, it seems worked. Will test again

**Comment by mjdousti** [comment]
I'm using PyTorch 2.0.1 with CUDA 11.8 and still face the same error. Note that I installed NVIDIA driver and CUDA via the binary which I got from NVIDIA website. The workaround I found was to install `cudatoolkit` via conda (it may also work without `-c pytorch -c nvidia` but I haven't tried it):
```
conda install cudatoolkit=11.8 -c pytorch -c nvidia

```
I think `cudatoolkit=11.8` should be added to the installation instruction provided on the PyTorch website.
